### DISCUSSION TITLE
table component search bar value

### DISCUSSION THREAD

**Discussion by ToniJ123** [sqlpage/SQLPage#645] [Ideas]
Hi everyone,

I'am creating web application which has card compononent in the site, there is two cards, on the left side card we have table component inside it as embed, and on the right side we have form component as embed. I've created three actions on the page:

1. For filtering data for table, start date and end date 
2. For transferring record's data to the right side's form
3. Form action as updating the selected record and returning to the site and telling did it went succesfully

![search_bar_value_in_table_component](https://github.com/user-attachments/assets/2990e22d-0527-44f7-b401-fb6fde6a9f2b)

Each of these refresh the site of course and when you have something searched from table's search bar, we lose this value everytime. Very annoying, as we can transfer every other data with us when doing anything. Of course we could transfer this value also in URL parameter, but at the moment we cannot affect any way to table component's search bar value. In html it is there easily modifiable.

![search_bar_value_in_table_component2](https://github.com/user-attachments/assets/42623c16-9c79-42e9-a142-9a39e9d57804)

So could this be added to the table component as parameter something like **search bar value**?

In our case it would be extremely useful, what do you fellow developers think, do you have anything like this, where this would be useful?

**Comment by lovasoa** [comment]
I'm adding a new `initial_search_value` attribute to the table component

**Comment by lovasoa** [comment]
I added  initial_search_value to the table component ! You are welcome to test the latest builds !

**Reply by lovasoa** [reply]
You can find the latest windows debug build here: https://send.vis.ee/download/c89963b78cb9b2f7/#znaNcHiJ4JqmbHgNGXW3oQ

**Reply by ToniJ123** [reply]
> I added initial_search_value to the table component ! You are welcome to test the latest builds !

Thanks this is great, although this is just an half of a solution, at least in our case.

So now you can define the value FOR the search bar. But we need to get also read the value FROM the search bar.
We are going to travel that value with us, so at the moment we cannot read what user has wrote to the search bar, so we
cannot predefine with this new initial_search_value attribute, the right value there, to keep the search bar value same, as it was
before having redirect visit in other site.

Could this be solved somehow? Thanks in advance!👍🙏🌞

**Reply by lovasoa** [reply]
Yes, but I don't think that belongs to the SQLPage core. You should probably include a short snippet of javascript that listens to input events in the search bar, and updates a hidden field in your form.
